### PR TITLE
Embeddability changes for Tietokilta's website

### LINKED ISSUE #16: Create integrable site components
For tietokilta.fi, we want to integrate the signup and event system with the main website. This will be accomplished by embedding dynamic parts into our otherwise-static website.

Required:
- Event list for front page
- Event list as standalone page (loaded inside blank page template)
- Event details (loaded inside blank page template)

Other ideas:
- Event list as calendar
- Event details for embedding on a page
- Pre-filtered event list (e.g. fuksi events on related page)
- Integrate admin pages as well?

How this will actually be implemented:
- Make parts of the components reusable: e.g. HOC for rendering an event list or event page
- Decouple styling from those parts enough that Bootstrap is not required (or necessary parts are prefixed)
- Potentially separate frontend build system into separate package, so we don't need to ship it everywhere

### LINKED ISSUE #85: Option to configure mailgun host
Currently `packages/ilmomasiina-backend/src/mail/config.ts` doesn't pass nodemailer-mailgun-transport the `host` config option. For MailGun domains that exist in the EU region one has to set that variable to `api.eu.mailgun.net` instead of the default `api.mailgun.net` for messages to be sent. Otherwise a vague 401 error is returned and shown in the logs with no context where it came from.

https://github.com/orliesaurus/nodemailer-mailgun-transport#mailgun-regions

### PR DESCRIPTION
**Pull request by PurkkaKoodari**
This PR will collect some changes that deemed necessary for integration with https://github.com/Tietokilta/tikweb-frontend, especially to the `ilmomasiina-components` package. This also contains some minor features that were implemented during the process.

Closes #16 - this produced components that allow multiple strategies for integration, including the one we're now using in tikweb-frontend.

### Features/changes

* Event descriptions and other descriptive fields are now also included in the event list response
* Signup counts are now displayed even if signups are hidden (this was already being returned by the API)
* Signup deletion no longer shows a dedicated page
* Signups can now be viewed even when they're no longer editable
* Added configurable CORS handling to the API
* The `EMAIL_BASE_URL` env variable is replaced with `BASE_URL` which also includes `PATH_PREFIX`. (This was made backwards compatible but prints a deprecation warning.)
* The backend now allows configuring the URLs generated for emails/iCalendar
* Mailgun host can be configured (closes #85)

### Refactorings

* Read paths from a React Context instead of a global
* Extract some reusable functionality to separate functions
* Email templates are now cleaner and handle e.g. branding changes better
* Updated Actions and made them use pnpm caching
* Fixed concurrency issues with pnpm + tsc
* Finally cleaning up the cross-package imports to be consistent. This was a pretty gigantic pain in the arse to figure out:
    * (1) The backend now actually imports compiled files from `ilmomasiina-models/dist`, so we don't need to mess with the paths in the Dockerfile and the compiled .js actually runs as-is
    * (2) To allow (1), the libraries are now referenced by `references` in tsconfig.json, so `tsc` can figure out that it can build the referenced files in `dist`
    * (3) Due to (2), we now need to use `tsc --build` for type-checking (due to https://github.com/microsoft/TypeScript/issues/40431)
    * (4) `ts-node` (and by extension `ts-node-dev`) don't understand `references`, so we need to use `paths` in tsconfig.json plus [tsconfig-paths](https://www.npmjs.com/package/tsconfig-paths) to tell them that our `dist` files are actually in `src`
    * (5) ESBuild also doesn't understand `references`. So, we need a plugin to resolve stuff. I wrote my own to do this using the same `paths` in tsconfig.json.
    * (6) Also, most imports now actually follow my own instructions and mostly import things from the "public" `index.ts` instead of grabbing stuff from `dist`
    * After an unreasonable number of hours I managed to scratch together a setup where each of these work at the same time, please don't break :~)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    "node": "^14.19.0",
-    "npm": "^6.14.13"
+    "node": "^16",
+    "npm": "^7"
   },
   "scripts": {
     "bootstrap": "pnpm install",
     "clean": "pnpm run -r clean",
-    "build": "pnpm run -r build",
+    "build": "pnpm run -r --workspace-concurrency=1 build",
```
Comment (epelip) [inline package.json:13]: Why is this required? I assume there is a good reason, but would be a good idea to document and write it down.

File: packages/ilmomasiina-backend/src/config.ts [hunk 2]
```diff
@@ -2,51 +2,109 @@ import {
...
+   */
+  baseUrl: envString('BASE_URL'),
+  /** URL template for an event details page. Used for iCalendar exports. Contains `{slug}`.
+   * @example "http://example.com/events/{slug}"
+   */
+  eventDetailsUrl: envString('EVENT_DETAILS_URL', `${envString('BASE_URL')}/events/{slug}`),
+  /** URL template for a signup edit page. Used for emails. Contains `{id}` and `{editToken}`.
+   * @example "http://example.com/signup/{id}/{editToken}"
+   */
+  editSignupUrl: envString('EDIT_SIGNUP_URL', `${envString('BASE_URL')}/signup/{id}/{editToken}`),
```
Comment (epelip) [inline packages/ilmomasiina-backend/src/config.ts]: At a first glance, these might seem redundant as the paths are hard-defined by the backend api. I assume these are added to provide support for customized frontends hosted from another location? Documenting the intended use case might be a good idea.

**Review by epelip — APPROVED** [review]
Looks good and functionally ready to be merged. Some changes and new features might need a bit more documentation (see the comments above), but those are more nitty-gritty details. Well done!

File: package.json (hunk 1 above)
Comment (PurkkaKoodari) [reply package.json:13]: Basically, I got lots of heisenbugs (with errors appearing out of nowhere) when multiple `tsc` instances were building the same folder simultaneously. This seems to eliminate that.

Comment (PurkkaKoodari) [reply package.json:13]: Unfortunately, `package.json` doesn't have many options for documentation...
